src/SharpCompress/IO/ReadOnlySubStream.cs

```diff
@@ -31,7 +31,7 @@ public ReadOnlySubStream(Stream stream, long? origin, long bytesToRead)
 
     public override void Flush() { }
 
-    public override long Length => throw new NotSupportedException();
```

Comment:
I don't think this is correct. The length of a substream is not equal to the length of the underlying `base` stream.

Reply:
actually, you're right.....I need a test or something to ensure things